python/lsst/ts/electrometer/controller.py

```diff
@@ -333,47 +355,120 @@ async def start_scan_dt(self, scan_duration):
+        primary_hdu.header["INSTRUME"] = (
+            f"Electrometer_index_{self.csc.salinfo.index}",
+            "Type of Instrument",
+        )
+        primary_hdu.header["MODEL"] = (self.model_id, "Model of instrument")
+        primary_hdu.header["LOCATN"] = (self.location, "Location of Instrument")
+        primary_hdu.header["ORIGIN"] = (
+            self.csc.salinfo.name,
+            "Name of the program that produced this data.",
+        )
+        primary_hdu.header["DATE-BEG"] = (
+            self.manual_start_time,
+            "When start scan command sent to CSC (TAI)",
+        )
+        primary_hdu.header["DATE-END"] = (
+            self.manual_end_time,
+            "When stop scan command sent to CSC (TAI)",
+        )
+        primary_hdu.header["TIMESYS"] = ("TAI", "Format of timestamps")
+        primary_hdu.header["SCANTIME"] = (self.scan_duration, "Duration of scan [s]")
+        primary_hdu.header["SAMPTIME"] = (
+            self.integration_time,
+            "Duration of each sample [s]",
+        )
+        primary_hdu.header["FILTMED"] = (
+            self.median_filter_active,
+            "Median Filter Active",
+        )
+        primary_hdu.header["FILTAVG"] = (
+            self.avg_filter_active,
+            "Average Filter Active",
+        )
+        primary_hdu.header["IMGTYPE"] = (
+            self.mode,
+            "Options are charge, voltage, current",
+        )
+        primary_hdu.header["SENSBRND"] = (self.sensor_brand, "Sensor brand")
+        primary_hdu.header["SENSMODL"] = (self.sensor_model, "Sensor model")
+        primary_hdu.header["SERIAL"] = (self.sensor_serial, "Sensor serial number")
+        primary_hdu.header["TEMP"] = (
+            self.temperature,
+            "Measurement from probe if attached and declared (Celcius)",
+        )
+        primary_hdu.header["VSOURCE"] = (
+            self.vsource,
+            "Voltage input if active and attached",
+        )
+        return primary_hdu
+
+    async def write_fits_file(self, signal, times, temperature, unit, voltage):
         """Write fits file of the intensity, time, and temperature values.
 
         Parameters
         ----------
-        intensity : `list`
-            The intensity values
-        times : `list`
-            The time of the intensity value
-        temperature : `list`
-            The temperature of the intensity value
-        unit : `list`
-            The unit of the intensity value.
+        signal : `list` of `float`
+            The amount of photons in a given reading, unit depends on mode of
+            electrometer.
+            * Curr: Ampere - Measure current
+            * Volt: V - Measure volts
+            * Char: Coulomb - Measure charge
+        times : `list` of `float`
+            The time (TAI) of the signal data taken.
+        temperature : `list` of `float`
+            The temperature (deg_C) of the signal data. (constant)
```

Comment:
By `constant` you mean a single value for the entire scan, yes?
If so, maybe consider:
A single measurement value from the temperature probe (deg_C) obtained during the signal acquisition. (constant)